src/test/java/org/apache/sling/resourceresolver/impl/mapping/VanityPathMapEntriesTest.java

```diff
@@ -1271,16 +1272,35 @@ public void test_bg_init_fallback_while_no_ready() {
         assertNotNull(mit);
         assertEquals(1, queryCounter.get());
 
+        // send a change event; this will be queud while init is running
+        // and then processed later on
+        Resource eventTest = mock(Resource.class, "eventTest");
+        when(eventTest.getName()).thenReturn("eventTest");
+        when(eventTest.getPath()).thenReturn("/eventTest");
+        when(eventTest.getValueMap()).thenReturn(buildValueMap("sling:vanityPath", "/baa"));
+        Resource eventTestTarget = mock(Resource.class, "eventTestTarget");
+        when(eventTestTarget.getName()).thenReturn("baa");
+        when(eventTestTarget.getPath()).thenReturn("/baa");
+        when(eventTestTarget.getValueMap()).thenReturn(buildValueMap("sling:vanityPath", "/baa"));
```

Comment:
I suggest that the ``sling:vanityPath`` differs from the actual path of that resource. Avoids ambiguity and shows potential issues in the code.